File: src/main/java/com/nurkiewicz/tsclass/parser/visitors/SourceElementVisitor.java
package com.nurkiewicz.tsclass.parser.visitors;

import com.nurkiewicz.tsclass.antlr.parser.TypeScriptBaseVisitor;
import com.nurkiewicz.tsclass.antlr.parser.TypeScriptParser;
import com.nurkiewicz.tsclass.parser.ast.Expression;
import com.nurkiewicz.tsclass.parser.ast.ReturnStatement;
import com.nurkiewicz.tsclass.parser.ast.Statement;

class SourceElementVisitor extends TypeScriptBaseVisitor<Statement> {

    @Override
    public Statement visitReturnStatement(TypeScriptParser.ReturnStatementContext ctx) {
        final Expression expression = ctx.expression().accept(new ExpressionVisitor());
        return new ReturnStatement(expression);
    }

}

File: src/main/kotlin/com/nurkiewicz/tsclass/parser/visitors/SourceElementVisitor.kt
package com.nurkiewicz.tsclass.parser.visitors

import com.nurkiewicz.tsclass.antlr.parser.TypeScriptBaseVisitor
import com.nurkiewicz.tsclass.antlr.parser.TypeScriptParser
import com.nurkiewicz.tsclass.parser.ast.ReturnStatement
import com.nurkiewicz.tsclass.parser.ast.Statement

internal class SourceElementVisitor : TypeScriptBaseVisitor<Statement>() {

    override fun visitReturnStatement(ctx: TypeScriptParser.ReturnStatementContext): Statement {
        val expression = ctx.expression().accept(ExpressionVisitor())
        return ReturnStatement(expression)
    }

}
